collector/diskstats_linux_test.go

```diff
@@ -14,39 +14,333 @@
 package collector
 
 import (
+	"fmt"
 	"os"
+	"strings"
 	"testing"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
 )
 
-func TestDiskStats(t *testing.T) {
-	file, err := os.Open("fixtures/proc/diskstats")
+type testDiskStatsCollector struct {
```

Comment:
Might make sense to have a package for this style of testing that works for other collectors as well, but that's all good for now I'd say.

Reply:
I realized I had mostly reimplemented testutil.GatherAndCompare(), so I dropped most of this code in favor of that. My plan is to refactor at least ethtool_linux_test to use this method, I'll include a common mechanism then. It doesn't belong in e.g. exporter-toolkit though because most of this code is just a wrapper for node_exporter's Collector type (`Update()`) to make it compatible with prometheus' Collector type (`Describe()` and `Collect()`).